src/renderer/dx/DxRenderer.cpp

```diff
@@ -1047,14 +1047,10 @@ HANDLE DxEngine::GetSwapChainHandle()
 
 void DxEngine::_InvalidateRectangle(const til::rectangle& rc)
 {
-    auto invalidate = rc;
-
-    if (_invalidateFullRows)
-    {
-        invalidate = til::rectangle{ til::point{ static_cast<ptrdiff_t>(0), rc.top() }, til::size{ _invalidMap.size().width(), rc.height() } };
-    }
-
-    _invalidMap.set(invalidate);
+    const auto size = _invalidMap.size();
+    const auto topLeft = til::point{ 0, std::min(size.height(), rc.top()) };
+    const auto bottomRight = til::point{ size.width(), std::min(size.height(), rc.bottom()) };
+    _invalidMap.set({ topLeft, bottomRight });

```

Comment:
Due to the missing `std::min`, but the delayed `_invalidMap` resize, this was blowing up with exceptions. Now it isn't anymore.